src/components/Calculator.js

```diff
@@ -1,43 +1,66 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      obj: {
+        total: null,
+        next: null,
+        operation: null,
+
+      },
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
+    const { obj } = this.state;
+    this.setState({
+      obj: calculate(obj, buttonName),
+    });
+  }
+
   render() {
+    const { obj } = this.state;
+
     return (
       <div className="calcMainContainer">
-        <div className="calcDisplay">0</div>
+        <div className="calcDisplay">{obj.next || obj.operation || obj.total || 0 }</div>
         <div className="calcKeysContainer">
           <div className="calcKeys">
             <div className="calcNumbers">
-              <button type="button">AC</button>
-              <button type="button">+/-</button>
-              <button type="button">%</button>
+              <button type="button" onClick={() => this.handleClick('AC')}>AC</button>
+              <button type="button" onClick={() => this.handleClick('+/-')}>+/-</button>
+              <button type="button" onClick={() => this.handleClick('%')}>%</button>
             </div>
             <div className="calcNumbers">
-              <button type="button">7</button>
-              <button type="button">8</button>
-              <button type="button">9</button>
+              <button type="button" onClick={() => this.handleClick('7')}>7</button>
+              <button type="button" onClick={() => this.handleClick('8')}>8</button>
+              <button type="button" onClick={() => this.handleClick('9')}>9</button>
             </div>
             <div className="calcNumbers">
-              <button type="button">4</button>
-              <button type="button">5</button>
-              <button type="button">6</button>
+              <button type="button" onClick={() => this.handleClick('4')}>4</button>
+              <button type="button" onClick={() => this.handleClick('5')}>5</button>
+              <button type="button" onClick={() => this.handleClick('6')}>6</button>
             </div>
             <div className="calcNumbers">
-              <button type="button">1</button>
-              <button type="button">2</button>
+              <button type="button" onClick={() => this.handleClick('1')}>1</button>
+              <button type="button" onClick={() => this.handleClick('2')}>2</button>
               <button type="button">3</button>
             </div>
```

Comment:
- [ ] You don't have an `onClick` function for **_3_**, so when I press 3, nothing is displayed. Kindly fix it. 👍